src/node_report_utils.cc

```diff
@@ -95,6 +95,8 @@ static void ReportPipeEndpoints(uv_handle_t* h, JSONWriter* writer) {
     buffer = MallocedBuffer<char>(buffer_size);
     if (buffer.data != nullptr) {
       rc = uv_pipe_getsockname(&handle->pipe, buffer.data, &buffer_size);
+    } else {
+      buffer_size = 0;
     }
```

Comment:
Actually, `buffer.data` cannot be `null` here because this ctor eventually calls `UncheckedRealloc<char>(nullptr, 1)` which returns a 1 byte buffer. Here's the inverted call stack:

https://github.com/nodejs/node/blob/be011858446f7ec4b914fd1af2cb7555bd04877b/src/util.h#L580

https://github.com/nodejs/node/blob/be011858446f7ec4b914fd1af2cb7555bd04877b/src/util-inl.h#L382-L387

https://github.com/nodejs/node/blob/be011858446f7ec4b914fd1af2cb7555bd04877b/src/util-inl.h#L361-L366

So this should be the proper fix:

```suggestion
    rc = uv_pipe_getsockname(&handle->pipe, buffer.data, &buffer_size);
```

Additionally, you could also add a DCHECK to make sure that the buffer is not null before calling `uv_pipe_getsockname()` since the function expects the caller to send in an allocated buffer:

```suggestion
    DCHECK(buffer.data);
    rc = uv_pipe_getsockname(&handle->pipe, buffer.data, &buffer_size);
```

Reply:
@RaisinTen 

But UncheckedRealloc can return a nullptr. It tries hard not to, but the final   `allocated = realloc(pointer, full_size);` could still fail right?
```
template <typename T>
T* UncheckedRealloc(T* pointer, size_t n) {
  size_t full_size = MultiplyWithOverflowCheck(sizeof(T), n);

  if (full_size == 0) {
    free(pointer);
    return nullptr;
  }

  void* allocated = realloc(pointer, full_size);

  if (UNLIKELY(allocated == nullptr)) {
    // Tell V8 that memory is low and retry.
    LowMemoryNotification();
    allocated = realloc(pointer, full_size);
  }

  return static_cast<T*>(allocated);
}
```